### PR TITLE
Verify GCP Task Library

### PR DESCRIPTION
**Pull request by ahuang11**
Consolidates
https://github.com/PrefectHQ/prefect/pull/5614
https://github.com/PrefectHQ/prefect/pull/5607
https://github.com/PrefectHQ/prefect/pull/5603

### CODE REVIEW

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks for consolidating these PRs! Code changes look good. One comment on the changes file and can you also make the changes to the docs so that the GCS tasks have the verified badge?

File: docs/core/task_library/overview.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Tasks marked with the <Badge text="Verified" type="success" vertical="middle"></
   <tr>
     <td><Badge text="Verified" type="success"></Badge><img src="/logos/fivetran.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/fivetran.html">Fivetran</a></td>
     <td><img src="/logos/github.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/github.html">GitHub</a></td>
-    <td><img src="/logos/google_cloud.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/gcp.html">Google Cloud</a></td>
+    <td><Badge text="Verified" type="success"><img src="/logos/google_cloud.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/gcp.html">Google Cloud</a></td>
```
Comment (desertaxle) [inline docs/core/task_library/overview.md:44]:
```suggestion
    <td><Badge text="Verified" type="success"></Badge><img src="/logos/google_cloud.png" height="128" width="128" style="max-height: 128px; max-width: 128px;"><a href="https://docs.prefect.io/api/latest/tasks/gcp.html">Google Cloud</a></td>
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Realized when fixing the Census display issue that we need to close the Badge tag here also.

**Review by tpdorsey — APPROVED** [review]

**Review by desertaxle — APPROVED** [review]

**Comment by ahuang11** [comment]
The tests passed, but GitHub isn't receiving the results.

**Pull request merged**